api/v1alpha2/vpcendpoint_types.go

```diff
@@ -0,0 +1,206 @@
+	SecurityGroup SecurityGroup `json:"securityGroup"`
+
+	// +kubebuilder:validation:Optional
+
+	// AssumeRoleArn will allow AVO to use sts:AssumeRole to create VPC Endpoints in separate AWS Accounts
+	// TODO: Implement
+	AssumeRoleArn string `json:"assumeRoleArn,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// Region will allow AVO to create VPC Endpoints and other AWS infrastructure in a specific region
+	// Defaults to the same region as the cluster AVO is running on
+	Region string `json:"region,omitempty"`
+
+	// +kubebuilder:default=false
+	// +kubebuilder:validation:Optional
+
+	// EnablePrivateDns will allow AVO to create VPC Endpoints with private DNS names specified by a VPC Endpoint Service
+	// https://docs.aws.amazon.com/vpc/latest/privatelink/manage-dns-names.html (defaults to false)
+	// TODO: Implement
+	EnablePrivateDns bool `json:"enablePrivateDns,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// Vpc will allow AVO to use a specific VPC or use the same VPC as the ROSA cluster it's running on
+	Vpc Vpc `json:"vpc,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// CustomDns will define configurations for all other custom DNS setups, such as a separate Route 53 Private Hosted
+	// Zone or an `ExternalName` Kubernetes service.
+	CustomDns CustomDns `json:"customDns,omitempty"`
+}
+
+const (
+	AWSVpcEndpointCondition      = "AWSVpcEndpointReady"
+	AWSSecurityGroupCondition    = "AWSSecurityGroupReady"
+	ExternalNameServiceCondition = "ExternalNameServiceReady"
+	AWSRoute53RecordCondition    = "AWSRoute53RecordReady"
+)
+
+// VpcEndpointStatus defines the observed state of VpcEndpoint
+type VpcEndpointStatus struct {
+	// Status of the VPC Endpoint
+	Status string `json:"status,omitempty"`
+
+	// The AWS ID of the managed security group
+	// +kubebuilder:validation:Optional
+	SecurityGroupId string `json:"securityGroupId,omitempty"`
+
+	// The AWS ID of the managed VPC Endpoint
+	// +kubebuilder:validation:Optional
+	VPCEndpointId string `json:"vpcEndpointId,omitempty"`
+
+	// The AWS ID of the Route 53 Private Hosted Zone being used
+	// +kubebuilder:validation:Optional
+	HostedZoneId string `json:"hostedZoneId,omitempty"`
+
+	// The FQDN of a Route 53 Hosted Zone record that has been created
+	// +kubebuilder:validation:Optional
+	ResourceRecordSet string `json:"resourceRecordSet,omitempty"`
+
+	// The status conditions of the AWS and K8s resources managed by this controller
+	// +kubebuilder:validation:Optional
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName={vpce},scope="Namespaced"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.status`
+// +kubebuilder:printcolumn:name="ID",type=string,JSONPath=`.status.vpcEndpointId`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:storageversion
```

Comment:
fyi this `// +kubebuilder:storageversion` is what makes the API server store v1alpha2 over v1alpha1